lib/pages/pool/MyLiqudity/MyLiquidity.dart

```diff
@@ -793,4 +796,34 @@ class _MyLiquidityState extends State<MyLiquidity> {
       ),
     );
   }
+
+  Widget badgeToken(SupportedSport sport, String symbol) {
+    return Badge(
+      shape: BadgeShape.square,
+      borderRadius: BorderRadius.circular(8),
+      badgeContent: Text(SupportedSportName[sport]!,
+          style: textStyle(Colors.white, 12, true)),
+      position: BadgePosition.topEnd(top: -14, end: -14),
+      padding: EdgeInsets.only(top: 2, bottom: 2, left: 5, right: 5),
+      child: Text(
+        symbol,
+        style: textStyle(Colors.white, 24, true),
+      ),
+    );
+  }
+
+  Widget simpleToken(String symbol) {
+    return Container(
+      alignment: Alignment.centerLeft,
+      child: Text(
+        symbol,
+        style: textStyle(Colors.white, 24, true),
+      ),
+    );
+  }
+
+  Widget sportToken(SupportedSport sport, String symbol) {
+    if (sport == SupportedSport.ALL) return simpleToken(symbol);
+    return badgeToken(sport, symbol);
+  }
```

Comment:
As per this: https://dartcodemetrics.dev/docs/rules/flutter/avoid-returning-widgets
We should avoid returning methods. Instead we should create separate widgets and call them when necessary.